packages/amplify-migration-tests/src/__tests__/migration_tests_v10/api-graphql-v1.migration.test.ts

```diff
@@ -0,0 +1,80 @@
+import { addApiWithoutSchema, 
+    addFunction, 
+    amplifyPull, 
+    amplifyPush, 
+    amplifyPushAuth, 
+    amplifyPushWithoutCodegen, 
+    createNewProjectDir, 
+    deleteProject, 
+    deleteProjectDir, 
+    generateRandomShortId, 
+    getAppId, 
+    updateApiSchema, 
+    updateFunction } from "@aws-amplify/amplify-e2e-core";
+import { initJSProjectWithProfileV10 } from "../../migration-helpers-v10/init";
+import { 
+    assertNoParameterChangesBetweenProjects, 
+    collectCloudformationDiffBetweenProjects, 
+} from "../../migration-helpers/utils";
+
+describe('api graphql v1 migration tests', () => {
+    let projRoot: string;
+    let projectName: string;
+
+    afterEach(async () => {
+        await deleteProject(projRoot, undefined, true);
+        deleteProjectDir(projRoot);
+    });
+
+    // inspired by function_5.test.ts
+    it('...adds graphql with v1 transformer, updates models, adds function access, and pulls with latest version', async () => {
+        projectName = 'graphqlMigration';
+        projRoot = await createNewProjectDir(projectName);
+        
+        await initJSProjectWithProfileV10(projRoot, { name: projectName, disableAmplifyAppCreation: false });
+        await addApiWithoutSchema(projRoot, { transformerVersion: 1 });
+        await updateApiSchema(projRoot, projectName, 'simple_model.graphql');
+        const fnName = `integtestfn${generateRandomShortId()}`;
+        await addFunction(
+            projRoot,
+            {
+                name: fnName,
+                functionTemplate: 'Hello World',
+            },
+            'nodejs',
+        );
+        await amplifyPush(projRoot);
+        await updateApiSchema(projRoot, projectName, 'two-model-schema.graphql');
+        await updateFunction(
+            projRoot,
+            {
+                name: fnName,
+                functionTemplate: 'Hello World',
+                additionalPermissions: {
+                permissions: ['storage'],
+                choices: ['api', 'function', 'storage'],
+                resources: ['Comment:@model(appsync)'],
+                resourceChoices: ['Post:@model(appsync)', 'Comment:@model(appsync)'],
+                operations: ['read'],
+                },
+            },
+            'nodejs',
+        );
+        await amplifyPushWithoutCodegen(projRoot, undefined, true);
```

Comment:
is pushing with latest codebase here expected ?

Reply:
this method is weird; the middle parameter is for testingWithLatestCodebase